### PR TITLE
remove `^` character to pin npm versions

### PR DESCRIPTION
**Pull request by anjbur**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Remove `^` characters to pin version in npm.

## Related PRs or issues

[AB#65909](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/65909)
https://github.com/microsoft/appcenter-sdk-react-native/pull/699

### CODE REVIEW

**Comment by guperrot** [comment]
```
> DemoApp@0.0.1 lint /Users/vsts/agent/2.155.1/work/1/s/DemoApp
> eslint .


/Users/vsts/agent/2.155.1/work/1/s/DemoApp/__mocks__/@react-native-community/async-storage.js
  1:1  error  Definition for rule 'react/jsx-curly-brace-presence' was not found  react/jsx-curly-brace-presence
```

**Comment by guperrot** [comment]
Might want to exclude TestApp and DemoApp for now for this change.

**Comment by YiMysty** [comment]
eslint --fix ? :D

File: TestAppTypescript/package.json [hunk 1]
```diff
@@ -12,27 +12,27 @@
     "clean": "rimraf artifacts"
   },
   "dependencies": {
-    "@react-native-community/async-storage": "^1.5.0",
```
Comment (guperrot) [inline TestAppTypescript/package.json]: Revert this one too then

**Comment by guperrot** [comment]
> eslint --fix ? :D

a PR build would not patch git though, needs to be done locally

**Pull request merged**